module-4/backend/week_two.md

```diff
@@ -1,22 +1,68 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
+
+  semi-colons aren't required in ES6.
+
+2. What's the difference between asynchronous and synchronous JavaScript?
+
+  Javascript in a single run-time environment is _synchronous_. A browser allows for Async because of the architecture of the environment.
+  This means in a browser environment you have to think about what will try to run when network requests are happening, and how you must account for this.
+
 3. What are the four pillars of Object Oriented programming?
+
+Abstraction
+Encapsulation
+Inheritance
+Polymorphism
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+
+Object contructors, `this`, and being able to change behavior with Object.prototype property.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+keep funcionality separated based on responsibility, especially in terms of things the do DOM manipulation, requests, things that deal with data persistence.
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+
+A Callback function is a functions that is triggered inside another function. We need to callbacks in EventHandlers, or generally when a function depends on something else happening first.
+
 7. What's the scope of variables in Javascript?
+
+`const` has global scope and is immutable
```

Comment:
not exactly `const` variable are still block scoped i.e if/else, loops. `const` and `let` variables can be both globally and locally/block scopes. I would discuss the difference between global and local variables